tools/SampleAnalyzer/Process/Plot/Histo.h

```diff
@@ -35,187 +33,142 @@
-
-    // Negative weight
-    else
-    {
-      nentries_.second++;
-      weight=std::abs(weight);
-      sum_w_.second   += weight;
-      sum_ww_.second  += weight*weight;
-      sum_xw_.second  += value*weight;
-      sum_xxw_.second += value*value*weight;
-      if (value < xmin_) underflow_.second+=weight;
-      else if (value >= xmax_) overflow_.second+=weight;
-      else
-      {
-        histo_[std::floor((value-xmin_)/step_)].second+=weight;
-      }
-    }
-  }
-
-  /// Write the plot in a ROOT file
-  virtual void Write_TextFormat(std::ostream* output);
-
-  /// Write the plot in a ROOT file
-  //  virtual void Write_RootFormat(std::pair<TH1F*,TH1F*>& histos);
-
- protected:
-
-  /// Write the plot in a ROOT file
-  virtual void Write_TextFormatBody(std::ostream* output);
-
-};
+        // -------------------------------------------------------------
+        //                        data members
+        // -------------------------------------------------------------
+    protected:
+        /// Each variable is defined with WEIGHTS object which includes positive and negative accessors
+        /// these are for positive and negative bins. std::map<MAint32,WEIGHTS> contains a map of
+        /// different PDF and their corresponding positive and negative weights.
+
+        /// Histogram arrays
+        std::vector<std::map<MAint32, WEIGHTS>> histo_;
+        std::map<MAint32, WEIGHTS> underflow_;
+        std::map<MAint32, WEIGHTS> overflow_;
+
+        /// Histogram description
+        MAuint32 nbins_;
+        MAfloat64 xmin_;
+        MAfloat64 xmax_;
+        MAfloat64 step_;
+
+        /// Sum of event-weights over entries
+        std::map<MAint32, WEIGHTS> sum_w_;
+
+        /// Sum of squared weights
+        std::map<MAint32, WEIGHTS> sum_ww_;
+
+        /// Sum of value * weight
+        std::map<MAint32, WEIGHTS> sum_xw_;
+
+        /// Sum of value * value * weight
+        std::map<MAint32, WEIGHTS> sum_xxw_;
+
+        /// RegionSelections attached to the histo
+        std::vector<RegionSelection *> regions_;
+
+        // -------------------------------------------------------------
+        //                       method members
+        // -------------------------------------------------------------
+    public:
+        /// Constructor without argument
+        Histo() : PlotBase()
+        {
+            nbins_ = 100;
+            xmin_ = 0;
+            xmax_ = 100;
+            step_ = (xmax_ - xmin_) / static_cast<MAfloat64>(nbins_);
+        }
+
+        /// Constructor with argument
+        Histo(const std::string &name) : PlotBase(name) {}
+
+        /// Constructor with argument
+        Histo(const std::string &name, MAuint32 nbins, MAfloat64 xmin, MAfloat64 xmax) : PlotBase(name)
+        {
+            // Setting the description: nbins
+            try
+            {
+                if (nbins == 0)
+                    throw EXCEPTION_WARNING("nbins cannot be equal to 0. Set 100.", "", 0);
```

Comment:
replace `Set 100` by `100 bins will be used`.

Reply:
```suggestion
                    throw EXCEPTION_WARNING("nbins cannot be equal to 0. 100 bins will be used.", "", 0);
```